src/components/reservation/ReservationForm.jsx

```diff
@@ -0,0 +1,188 @@
+  }, [yachts, dispatch]);
+
+  return (
+    <div className='p-4'>
+      <form onSubmit={handleSubmit} className='max-w-sm mx-auto'>
+        <div className='mb-4'>
+          <label
+            htmlFor='userName'
+            className='block text-gray-700 font-semibold mb-2'
+          >
+            {' '}
+            User Name
+            <input
+              type='text'
+              id='userName'
+              value={user.name}
+              disabled
+              className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+            />
+          </label>
+        </div>
+
+        {yId < 1 ? (
+          <div className='mb-4'>
+            <label
+              htmlFor='yachtName'
+              className='block text-gray-700 font-semibold mb-2'
+            >
+              Yacht Name
+              <input
+                type='text'
+                id='yachtName'
+                value={yachtName}
+                disabled
+                className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+              />
+            </label>
+          </div>
+        ) : (
+          <div className='mb-4'>
+            <label
+              htmlFor='yachtName'
+              className='block text-gray-700 font-semibold mb-2'
+            >
+              Yacht Name
+              <select
+                id='yachtName'
+                value={yachtId}
+                onChange={(e) => setYachtId(e.target.value)}
+                required
+                className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+              >
+                <option value=''>Select Yacht Name</option>
+                {yachts.map((yacht) => (
+                  <option key={yacht.id} value={yacht.id}>
+                    {yacht.model}
+                  </option>
+                ))}
+              </select>
+            </label>
+          </div>
+        )}
+
+        <div className='mb-4'>
+          <label
+            htmlFor='date'
+            className='block text-gray-700 font-semibold mb-2'
+          >
+            Date
+            <input
+              type='date'
+              id='date'
+              value={date}
+              onChange={(e) => setDate(e.target.value)}
+              required
+              className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+            />{' '}
+          </label>
+        </div>
+
+        <div className='mb-4'>
+          <label
+            htmlFor='city'
+            className='block text-gray-700 font-semibold mb-2'
+          >
+            City
+            <input
+              type='text'
+              id='city'
+              value={city}
+              onChange={(e) => setCity(e.target.value)}
+              required
+              className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+            />{' '}
+          </label>
```

Comment:
[OPTIONAL] Consider valdiating city chosen. I should not be able to put xyz. Maybe there is an API yo ucan use to validate cities or another way?